pilot/cmd/pilot-agent/config/config.go

```diff
@@ -73,6 +74,15 @@ func ConstructProxyConfig(meshConfigFile, serviceCluster, proxyConfigEnv string,
 		log.Warnf("legacy --concurrency=%d flag detected; prefer to use ProxyConfig", concurrency)
 		proxyConfig.Concurrency = wrapperspb.Int32(int32(concurrency))
 	}
+
+	if proxyConfig.Concurrency.GetValue() == 0 {
```

Comment:
How could it be 0, In L69 we set proxyConfig.Concurrency = wrapperspb.Int32(int32(CPULimit))

Reply:
Look at L64. This for if proxyConfig.Concurrency == nil